### PR TITLE
Show docs on kb selfservice

### PR DESCRIPTION
**Pull request by btry**
Internal ref: 21539

image

image

### CODE REVIEW

File: inc/document_item.class.php [hunk 1]
```diff
@@ -755,54 +755,9 @@ static function showListForItem(CommonDBTM $item, $withtemplate = 0, $options =
...
-            "$sort $order"
-         ]
-      ];
-
-      if (Session::getLoginUserID()) {
-         $criteria['WHERE'] = $criteria['WHERE'] + getEntitiesRestrictCriteria('glpi_documents', '', '', true);
-      } else {
-         // Anonymous access from FAQ
-         $criteria['WHERE']['glpi_documents.entities_id'] = 0;
-      }
```
Comment (btry) [inline inc/document_item.class.php:802]: the above request is also needed for  the PR, then I isolated it in a dedicated method for factorization need.

File: inc/document_item.class.php [hunk 2]
```diff
@@ -755,54 +755,9 @@ static function showListForItem(CommonDBTM $item, $withtemplate = 0, $options =
...
+      $criteria = self::getDocumentForItemRequest($item, ["$sort $order"]);
 
       // Document : search links in both order using union
-      $doc_criteria = [];
```
Comment (btry) [inline inc/document_item.class.php:805]: unused var : cleanup

File: inc/document_item.class.php [hunk 3]
```diff
@@ -888,7 +843,6 @@ static function showListForItem(CommonDBTM $item, $withtemplate = 0, $options =
                Session::addToNavigateListItems('Document', $docID);
             }
             $used[$docID] = $docID;
-            $assocID      = $data["assocID"];
```
Comment (btry) [inline inc/document_item.class.php:891]: unused var : cleanup again

File: inc/knowbaseitem.class.php [hunk 4]
```diff
@@ -957,6 +957,55 @@ function showFull($options = []) {
       $out.= "</div>";
       $out.= "</td></tr>";
 
+      // Show documents attached to the FAQ Item
+      $sort = 'name';
+      $order = 'ASC';
+      $criteria = Document_Item::getDocumentForItemRequest($this, ["$sort $order"]);
```
Comment (btry) [inline inc/knowbaseitem.class.php:963]: call to factorized code

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

File: inc/knowbaseitem.class.php [hunk 5]
```diff
@@ -957,6 +957,55 @@ function showFull($options = []) {
...
+      $iterator = $DB->request($criteria);
+      if (count($iterator) > 0) {
+         $columns = [
```
Comment (cedric-anne) [inline inc/knowbaseitem.class.php:966]:
```suggestion
         $out.= "<tr><td class='left' colspan='4'><h2>".Document::getTypeName(Session::getPluralNumber())."</h2></td></tr>\n";

         $columns = [
```

File: inc/knowbaseitem.class.php [hunk 6]
```diff
@@ -957,6 +957,57 @@ function showFull($options = []) {
...
+         $out .= $header_begin.$header_top.$header_end;
+
+         $document = new Document();
+         foreach ($iterator as $data) {
+            $docID        = $data["id"];
+            $link         = NOT_AVAILABLE;
+            $downloadlink = NOT_AVAILABLE;
+
+            if ($document->getFromDB($docID)) {
+               $link         = $document->getLink();
```
Comment (orthagh) [inline inc/knowbaseitem.class.php:994]: is, from a self-service pov, required to have a link to the glpi document.
I think the targeted page is not available to this kind of profiles

Comment (btry) [reply inc/knowbaseitem.class.php:994]: I did a test with a self service account. I can access a .png file from the link.

image


image

Comment (orthagh) [reply inc/knowbaseitem.class.php:994]: No, i talk about the document link not the download link.

But last week we also talked about having a name and a file name is redundant
Please show the link as
```
<a href="download_link">document name</a>
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

File: inc/knowbaseitem.class.php (hunk 6 above)
Comment (btry) [reply inc/knowbaseitem.class.php]: redundancy removed, also filtered out deleted files.

**Pull request merged**